vignettes/ae-exp-adj.Rmd

```diff
@@ -0,0 +1,347 @@
+                         tools::toTitleCase(tolower(adae$AEOUT[i]))
+                         )
+}
+
+# AE action
+adae$AEACN <- gsub("", "DOSE NOT CHANGED", adae$AEACN)
+
+for (i in seq_along(adae$AEACN)) {
+  adae$action_taken[i] <- switch(adae$AEACN[i],
+                                 "DOSE NOT CHANGED" = "None",
+                                 "DOSE REDUCED" = "Reduced",
+                                 "DRUG INTERRUPTED" = "Interrupted",
+                                 "DOSE INCREASED" = "Increased",
+                                 "NOT APPLICABLE" = "N/A",
+                                 "UNKNOWN" = "Unknown",
+                                 "''" = "None",
+                                 tools::toTitleCase(tolower(adae$AEACN[i]))
+                                 )
+}
+
+# AE duration with unit
+adae$duration <- paste(
+  ifelse(
+    is.na(adae$ADURN),
+    "",
+    as.character(adae$ADURN)
+    ),
+  tools::toTitleCase(tolower(adae$ADURU)),
+  sep = " "
+)
+
+for (i in seq_along(adae$duration)) {
+  if (is.na(adae$ADURN[i])) {
+    adae$duration[i] <- ifelse(
+      charmatch(toupper(adae$AEOUT[i]), "RECOVERING/RESOLVING") > 0 |
+        charmatch(toupper(adae$AEOUT[i]), "NOT RECOVERED/NOT RESOLVED") > 0,
+      "Continuing",
+      "Unknown"
+      )
+  }
+}
+
+# Assign label
+adae <- metalite::assign_label(adae,
+                               var = c("related", "outcome", "duration", "AESEV", "AESER", "AEDECOD", "action_taken"),
+                               label = c("Related", "Outcome", "Duration", "Intensity", "Serious", "Adverse Event", "Action Taken")
+)
+```
+At the end of this step, adae is created.(See .Rmd file for codes)
+```{r,include=FALSE}
+head(adae)
+```
+
+####  2. Build a metadata 
+
+```{r,echo=FALSE,warning=FALSE, message=FALSE}
+plan <- plan(
+    analysis = "ae_exp_adj", population = "apat",
+    observation = c("wk12", "wk24"), parameter = "any;rel;ser") 
+
+meta <- meta_adam(
+    population = adsl,
+    observation = adae
+    ) |>
+  define_plan(plan = plan) |>
+  define_population(
+    name = "apat",
+    group = "TRTA",
+    subset = quote(SAFFL == "Y"),
+    var = "TRTDUR"
+    ) |>
+  define_observation(
+    name = "wk12",
+    group = "TRTA",
+    subset = quote(SAFFL == "Y"),
+    label = "Weeks 0 to 12"
+  ) |>
+  define_observation(
+    name = "wk24",
+    group = "TRTA",
+    subset = quote(AOCC01FL == "Y"), # Just for demo, another flag should be used
+    label = "Weeks 0 to 24"
+  ) |>
+  define_parameter(
+    name = "rel",
+    subset = quote(AEREL %in% c("POSSIBLE", "PROBABLE"))
+  ) |>
+  define_analysis(
+    name = "ae_exp_adj",
+    title = "Exposure-Adjusted Adverse Event Summary"
+  ) |>
+  meta_build()
+```
+At the end of this step, 'meta' is created.(See .Rmd file for codes)
+```{r,include=FALSE}
+meta
+```
+
+
+#### 3.Call function prepare_ae_exp_adj() --- Function updated by Bing (--To Be Discussed.)

```

Comment:
In general, feel we can separate function development and vignettes creation into two PR. 
 
Before we merge to main branch, 

Could we move information below to either an issue or a working branch?